src/segment-loader.js

```diff
@@ -1048,10 +1083,8 @@ export default class SegmentLoader extends videojs.EventTarget {
         0, 0, 0, segmentInfo.mediaIndex + segmentInfo.playlist.mediaSequence
       ]);
 
-      simpleSegment.key = {
-        resolvedUri: segment.key.resolvedUri,
-        iv
-      };
+      simpleSegment.key = this.segmentKey(segment.key);
+      simpleSegment.key.iv = iv;
```

Comment:
This winds up adding the `iv` value to the cached key because `segmentKey` will return the reference

Reply:
Good catch. I don't think this will cause any issues as we don't ever use the "cached" iv value since we are always overwriting it here anyway. I can update this to return a new reference though